src/expr/impl/src/scalar/jsonb_contains.rs

```diff
@@ -0,0 +1,248 @@
+/// query B
+/// SELECT '"foo"'::jsonb @> '"foo"'::jsonb;
+/// ----
+/// t
+///
+/// # The array on the right side is contained within the one on the left:
+/// query B
+/// SELECT '[1, 2, 3]'::jsonb @> '[1, 3]'::jsonb;
+/// ----
+/// t
+///
+/// # Order of array elements is not significant, so this is also true:
+/// query B
+/// SELECT '[1, 2, 3]'::jsonb @> '[3, 1]'::jsonb;
+/// ----
+/// t
+///
+/// # Duplicate array elements don't matter either:
+/// query B
+/// SELECT '[1, 2, 3]'::jsonb @> '[1, 2, 2]'::jsonb;
+/// ----
+/// t
+///
+/// # The object with a single pair on the right side is contained
+/// # within the object on the left side:
+/// query B
+/// SELECT '{"product": "PostgreSQL", "version": 9.4, "jsonb": true}'::jsonb @> '{"version": 9.4}'::jsonb;
+/// ----
+/// t
+///
+/// # The array on the right side is not considered contained within the
+/// # array on the left, even though a similar array is nested within it:
+/// query B
+/// SELECT '[1, 2, [1, 3]]'::jsonb @> '[1, 3]'::jsonb;
+/// ----
+/// f
+///
+/// # But with a layer of nesting, it is contained:
+/// query B
+/// SELECT '[1, 2, [1, 3]]'::jsonb @> '[[1, 3]]'::jsonb;
+/// ----
+/// t
+///
+/// # Similarly, containment is not reported here:
+/// query B
+/// SELECT '{"foo": {"bar": "baz"}}'::jsonb @> '{"bar": "baz"}'::jsonb;
+/// ----
+/// f
+///
+/// # A top-level key and an empty object is contained:
+/// query B
+/// SELECT '{"foo": {"bar": "baz"}}'::jsonb @> '{"foo": {}}'::jsonb;
+/// ----
+/// t
+///
+/// # This array contains the primitive string value:
+/// query B
+/// SELECT '["foo", "bar"]'::jsonb @> '"bar"'::jsonb;
+/// ----
+/// t
+///
+/// # This exception is not reciprocal -- non-containment is reported here:
+/// query B
+/// SELECT '"bar"'::jsonb @> '["bar"]'::jsonb;
+/// ----
+/// f
+/// ```
+#[function("jsonb_contains(jsonb, jsonb) -> boolean")]
+fn jsonb_contains(left: JsonbRef<'_>, right: JsonbRef<'_>) -> bool {
+    jsonbb_contains(left.into(), right.into())
+}
+
+fn jsonbb_contains(left: ValueRef<'_>, right: ValueRef<'_>) -> bool {
+    match (left, right) {
+        // Both left and right are objects.
+        (ValueRef::Object(left_obj), ValueRef::Object(right_obj)) => {
+            // Every key-value pair in right should be present in left.
+            right_obj.iter().all(|(key, value)| {
+                left_obj
+                    .get(key)
+                    .map_or(false, |left_val| jsonbb_contains(left_val, value))
+            })
+        }
+
+        // Both left and right are arrays.
+        (ValueRef::Array(left_arr), ValueRef::Array(right_arr)) => {
+            // For every value in right, there should be an equivalent in left.
+            right_arr
+                .iter()
+                .all(|right_val| left_arr.iter().any(|left_val| left_val == right_val))
+        }
+
+        // Left is an array and right is a primitive value.
+        (ValueRef::Array(left_arr), right_val) => {
```

Comment:
`right_val` may not be a primitive here. The following is `false` in PostgreSQL but `true` in this PR:
```
select '[1, {"a":2}]'::jsonb @> '{"a":2}';
```